src/metabase/models/setting.clj

```diff
@@ -271,9 +271,10 @@
    ;; where this setting should be visible (default: :admin)
    [:visibility Visibility]
 
-   ;; should this setting be encrypted `:never` or `:maybe` (when `MB_ENCRYPTION_SECRET_KEY` is set).
-   ;; Defaults to `:maybe` (except for `:boolean` typed settings, where it defaults to `:never`)
-   [:encryption [:enum :never :maybe]]
+   ;; should this setting be encrypted. Available options are `:no` or `:when-encryption-key-set` (the setting will be
+   ;; encrypted when `MB_ENCRYPTION_SECRET_KEY` is set, otherwise we can't encrypt). This is required for `:timestamp`,
+   ;; `:json`, and `:csv`-typed settings. Defaults to `:no` for all other types.
+   [:encryption [:enum :no :when-encryption-key-set]]
```

Comment:
> For settings that are not typed as JSON, CSV, or strings, encryption now defaults to :never rather than :maybe (except if you have explicitly marked your setting as :sensitive? - these will default to :maybe).

I think we mean `:never -> :no` and `:maybe -> :when-encryption-key-set`. I'll edit the description.

Reply:
Yep, thank you!